core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -109,7 +109,7 @@ class AddPartitionsToTxnManager(
         .setTransactionalId(transactionalId)
         .setProducerId(producerId)
         .setProducerEpoch(producerEpoch)
-        .setVerifyOnly(true)
+        .setVerifyOnly(supportedOperation != addPartition)
```

Comment:
nit -- we should also add the supportedOperation in KafkaApis and/or the other files based on the request version.

Reply:
Reverted the kafka api changes, anywhere else needs to use supportedOperation?